src/cmd/ksh93/sh/io.c

```diff
@@ -1492,12 +1493,14 @@ int sh_redirect(Shell_t *shp, struct ionod *iop, int flag) {
                     if (fd < 10) {
                         if ((fn = fcntl(fd, F_DUPFD, 10)) < 0) goto fail;
                         if (fn >= shp->gd->lim.open_max && !sh_iovalidfd(shp, fn)) goto fail;
-                        if (flag != 2 || shp->subshell)
-                            sh_iosave(shp, fn, indx | 0x10000, tname ? fname : (trunc ? Empty : 0));
                         shp->fdstatus[fn] = shp->fdstatus[fd];
                         sh_close(fd);
                         fd = fn;
                     }
+
+                    if (flag != 2 || shp->subshell)
```

Comment:
Any idea what the magic number `2` is? Is it meant to be `IOWRITE`?

Reply:
Never mind. I see from the function comment that these are magic constants without associated symbols.